### PR TITLE
Feature/apc token storage

### PR DESCRIPTION
**Pull request by fedemotta**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,2 +1,3 @@
 vendor
 composer.lock
+/nbproject/
```
Comment (f3ath) [inline .gitignore:3]: What is this for?

File: src/ApcTokenStorage.php [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+ * @author Federico Nicolás Motta <fedemotta@gmail.com>
+ * @license http://opensource.org/licenses/mit-license.php The MIT License (MIT)
+ */
+class ApcTokenStorage implements TokenStorage
+{
+    private $prefix;
+    private $ttl;
+
+    public function __construct($prefix = '',  $ttl = 0 )
+    {
```
Comment (f3ath) [inline src/ApcTokenStorage.php:20]: I suggest

```
    if ( ! extension_loaded('apc')) {
        throw new \RuntimeException('APC extension is not loaded');
    }
```

File: .gitignore (hunk 1 above)
Comment (fedemotta) [inline .gitignore:3]: To ignore netbeans files. You should complete with other common rules for PhpStom, Netbeans, Windows thumbnail cache.  See https://github.com/yiisoft/yii2/blob/master/.gitignore as an example.

File: test/ApcTokenStorageTest.php [hunk 3]
```diff
@@ -0,0 +1,59 @@
+<?php
+
+namespace {
```
Comment (f3ath) [inline test/ApcTokenStorageTest.php]: I believe this global namespace is redundant, PHP does not support "namespaced" variables.

File: src/ApcTokenStorage.php (hunk 2 above)
Comment (fedemotta) [inline src/ApcTokenStorage.php:20]: Added in 8d0690286ce74021fbad65c362ab8c3fdd0e8d29

File: .gitignore (hunk 1 above)
Comment (f3ath) [inline .gitignore:3]: Ah, got it.

File: test/ApcTokenStorageTest.php (hunk 3 above)
Comment (fedemotta) [inline test/ApcTokenStorageTest.php]: This was the only way I found to test with apc. Apc didn't work on cli, so I need to mock the apc_store and apc_fetch functions

Comment (fedemotta) [inline test/ApcTokenStorageTest.php]: See this: http://marcelog.github.io/articles/php_mock_global_functions_for_unit_tests_with_phpunit.html

Comment (f3ath) [inline test/ApcTokenStorageTest.php]: Yes, I understand you're mocking the functions within the namespace. My comment was about declaring

```
$mockApcStore = false;
$mockApcFetch = false;
```

in the global namespace.  The vars can be declared in the F3\AppNexusClient namespace and will be still visible globally. I mean you don't have to have two namespaces in the file.

Comment (fedemotta) [inline test/ApcTokenStorageTest.php]: I see. Moved to F3\AppNexusClient namespace in 927308ea9a8c29727269a002c2601968084b7ac8

**Comment by f3ath** [comment]
Thank you. Version 0.1.0 is now available at  https://packagist.org/packages/f3ath/appnexus

**Comment by fedemotta** [comment]
Thank you for your work!

**Pull request merged**